src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -3041,7 +3041,9 @@ public override BoundNode VisitConditionalReceiver(BoundConditionalReceiver node
 
         public override BoundNode VisitLoweredIsPatternExpression(BoundLoweredIsPatternExpression node)
         {
+            var savedState = this.State.Clone();
             VisitStatements(node.Statements);
+            this.State = savedState;
             return null;
         }
```

Comment:
My reasoning for the change is that the lowered statements should not affect the state for anything that follows, therefore we start with initial state after visiting those statements. I have no way of verifying it though except for passing the repro. @RikkiGibson @AlekseyTs

Reply:
It looks like capture walker is the only walker that operates on the lowered bound tree and therefore seeing this node? Is it possible to always run this post-rewrite and assert that the results "match"? Just a way to exercise this code path more than usual.